salt/client/ssh/__init__.py

```diff
@@ -1393,6 +1393,28 @@ def categorize_shim_errors(self, stdout_bytes, stderr_bytes, retcode):
         perm_error_fmt = 'Permissions problem, target user may need '\
                          'to be root or use sudo:\n {0}'
 
+        def _version_mismatch_error():
+            messages = {
+                2: {
+                    6: 'Install Python 2.7 / Python 3 Salt dependencies on the Salt SSH master \n'
+                       'to interact with Python 2.7 / Python 3 targets',
+                    7: 'Install Python 2.6 / Python 3 Salt dependencies on the Salt SSH master \n'
+                       'to interact with Python 2.6 / Python 3 targets',
+                },
+                3: {
+                    'default': '- Install Python 2.6/2.7 Salt dependencies on the Salt SSH \n'
+                               '  master to interact with Python 2.6/2.7 targets\n'
+                               '- Install Python 3 on the target machine(s)',
+                },
+                'default': 'Matching major/minor Python release (>=2.6) needed both on the Salt SSH \n'
+                           'master and target machine',
+            }
+            major, minor = sys.version_info[:2]
+            help_msg = messages.get(major, {}).get(minor) \
+                or messages.get(major, {}).get('default') \
+                or messages['default']
```

Comment:
I would remove that ~ugly~ backslash here, as per [line continuation style guide](http://docs.python-guide.org/en/latest/writing/style/#line-continuations), but it is up to you.